### DISCUSSION TITLE
Index: converting string.Join('_' wrong result

### DISCUSSION THREAD

**Discussion by bwmeyer** [ravendb/ravendb#14032] [Q&A]
As the highlighting of double quotes in the studios index detail is wrong/misleading, I wanted to use single quotes.
So I have changed the call of the string.Join method accordingly.
Unfortunately this will not be correctly converted.

When an index with string.Join character based is created in c# it will be converted to this:
`String.Join < string > ('_',  ...`
instead of 
`String.Join('_', ...`

```
        [TestMethod]
        public void CreateIndexTest()
        {
            using var store = GetDocumentStore();
            new TestDocIndex().Execute(store);
            var index = store.Maintenance.Send(new GetIndexOperation("TestDocIndex"));
            Assert.IsFalse(index.Maps.First().Contains("String.Join < string > ("));
        }

        public class TestDoc : DocumentBase
        {
            public string Id { get; set; }
            public List<string> Data { get; set; } = new List<string>();
        }

        public class TestDocIndex : AbstractIndexCreationTask<TestDoc>
        {
            public TestDocIndex()
            {
                Map = testDocs => from testDoc in testDocs
                                  select new
                                  {
                                      testDoc.Id,
                                      Data = string.Join('_', testDoc.Data)
                                  };
                StoreAllFields(FieldStorage.Yes);
            }
        }``
```

**Comment by maciejaszyk** [comment]
Hi!
I've looked into that and found out it is a bug related to our index builder. I will take a look into that. As a workaround you can use a `string` as separator instead of `char` in `string.Join` and everything works just fine.

Thanks for your feedback.

```
public class TestDocIndex : AbstractIndexCreationTask<TestDoc>
    {
        public TestDocIndex()
        {
            Map = testDocs => from testDoc in testDocs
                select new
                {
                    testDoc.Id,
                    Data = string.Join("_", testDoc.Data) //or you can use LINQ: testDoc.Data.Aggregate(((s, s1) => $"{s}_{s1}"))
                };
            StoreAllFields(FieldStorage.Yes);
        }
    }

**Reply by bwmeyer** [reply]
This is what I have done, but the double quotes aren't highlighted well in studio.
<img width="563" alt="image" src="https://user-images.githubusercontent.com/39671001/162427016-88080d0d-3c57-4a09-945d-4c2a05b29c96.png">

**Reply by maciejaszyk** [reply]
Maybe this will help you.
`char.ToString('_')`

**Reply by arekpalinski** [reply]
@bwmeyer Okay so it's just the UI issue, right? 

@ml054 please have a look.

**Reply by bwmeyer** [reply]
Yes, the double quotes is only a UI issue (and was misleading me first)

**Reply by ml054** [reply]
@bwmeyer I've opened issue for that: https://issues.hibernatingrhinos.com/issue/RavenDB-18542  You can watch this issue to track progress.
